code/game/objects/items/crayons.dm

```diff
@@ -851,14 +857,34 @@
 				to_chat(user, span_warning("A color that dark on an object like this? Surely not..."))
 				return FALSE
 
-			target.add_atom_colour(paint_color, WASHABLE_COLOUR_PRIORITY)
 			if(isitem(target) && isliving(target.loc))
 				var/obj/item/target_item = target
 				var/mob/living/holder = target.loc
 				if(holder.is_holding(target_item))
 					holder.update_held_items()
 				else
 					holder.update_clothing(target_item.slot_flags)
+
+			if(istype(target, /obj/machinery/atmospherics))
+				var/obj/machinery/atmospherics/target_pipe = target
+				if(GLOB.pipe_color_name.Find(paint_color))
+					target_pipe.paint(paint_color)
+					balloon_alert(user, "painted in  [GLOB.pipe_color_name[paint_color]] color")
+				else
+					balloon_alert(user, "invalid pipe color!")
+					return FALSE
+			else if(istype(target, /obj/item/pipe))
+				var/obj/item/pipe/target_pipe = target
+				if(GLOB.pipe_color_name.Find(paint_color))
+					target_pipe.pipe_color = paint_color
+					target.add_atom_colour(paint_color, FIXED_COLOUR_PRIORITY)
+					balloon_alert(user, "painted in [GLOB.pipe_color_name[paint_color]] color")
+				else
+					balloon_alert(user, "invalid pipe color!")
+					return FALSE
+			else
+				target.add_atom_colour(paint_color, WASHABLE_COLOUR_PRIORITY)
```

Comment:
Hold on, putting it here will mean that visuals of worn/held items won't be updated correctly when painted. Move all of this before the items on.